docs/source/reference/envs.rst

```diff
@@ -169,7 +186,37 @@ one can simply call:
         >>> print(a)
         9.81
 
-It is also possible to reset some but not all of the environments:
+TorchRL uses a private ``"_reset"`` key to indicate to the environment which
+sub-environments should be reset.
+This allows to reset some but not all of the environments.
+Keep in mind that this is a private key, and it should only be used when coding
+specific environment features that are internal facing.
+The way an environment deals with the ``"_reset"`` keys is proper to it and
+designing an environment that behaves according to ``"_reset"`` inputs is the
+developer's responsibility. The following assumptions are made:
+
+- Each ``"_reset"`` is paired with a ``"done"`` (and/or ``"terminated"``,
+  ``"truncated"`` etc.). This means that the following structure is not
+  allowed: ``TensorDict({"done": done, "nested": {"_reset": reset}}, [])``.
+- A reset at one level precludes the presence of a ``"_reset"`` at lower
+  levels below it. For instance, this is not allowed:
+  ``TensorDict({"_reset": reset_main, "nested": {"_reset": reset_leaf}}, [])``
+- When calling :meth:`env.reset(tensordict)` with a partial ``"_reset"`` entry
+  that will reset some but not all the done sub-environments, the input data
+  should contain the data of the sub-environemtns that are __not__ being reset.
```

Comment:
How can the input contain only the data "of the sub-environments that are __not__ being reset." It has to contain all data no? otherwise it is sparse

Reply:
who said only?

Reply:
but if we specify that it should contain the data for those that are not being reset it might be read as implying that it contains only that. I think it is easier if we say it contains all data

Reply:
if we want to be precise, we don't really care that the data of the envs being reset is there actually. As long as `torch.where` will work.
I don't really see how it can contain one and not the other (without recurring to lazy stacks or such -- and if it does contain lazy stacks with just the non-reset data it should still work).

Reply:
Example:
```python
import torch
from tensordict import TensorDict
reset = torch.tensor([True, False])
td = torch.stack([
    TensorDict({}, []),
    TensorDict({"a": 1}, []),
])
td_reset = torch.stack([
    TensorDict({"a": 0}, []),
    TensorDict({}, []),
])
td_reset.where(~reset, other=td, pad=0)
```
which returns a non-sparse td
```
LazyStackedTensorDict(
    fields={
        a: Tensor(shape=torch.Size([2]), device=cpu, dtype=torch.int64, is_shared=False)},
    exclusive_fields={
    },
    batch_size=torch.Size([2]),
    device=None,
    is_shared=False,
    stack_dim=0)
```